Incorrect stats in exported CDRs (nondeterministic)

When exporting CDRs via ExportCDRs function sometime statistics are not correctly computed.

It can be also seen in unit tests sometime:
--- FAIL: TestWriteCdrs (0.00s)
    cdrefwv_test.go:257: Unexpected total cost in the stats: 3.65001
    cdrefwv_test.go:260: Unexpected FirstOrderId 3
    cdrefwv_test.go:266: Unexpected TotalCost:  3.65001

--- a/CONTRIBUTORS.md
+++ b/CONTRIBUTORS.md
@@ -68,6 +68,7 @@ information, please see the [`CONTRIBUTING.md`](CONTRIBUTING.md) file.
 | @petekelly | Peter Kelly |
 | @vtzan | Vasilios Tzanoudakis |
 | @ccppprogrammer | Sergei Lavrov |
+| @chrisgis333 | Krzysztof Grzyb |
 
 <!-- to sign, include a single line above this comment containing the following text:
 | @username | First Last |

--- a/engine/cdre.go
+++ b/engine/cdre.go
@@ -331,6 +331,8 @@ func (cdre *CDRExporter) processCDR(cdr *CDR) (err error) {
 		return
 	}
 	// Done with writing content, compute stats here
+	cdre.Lock()
+	defer cdre.Unlock()
 	if cdre.firstCdrATime.IsZero() || cdr.AnswerTime.Before(cdre.firstCdrATime) {
 		cdre.firstCdrATime = cdr.AnswerTime
 	}

